bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/services/EvaluationService.java

```diff
@@ -130,7 +130,7 @@ public void sourceChanged(int sourcePriority, Map sourceValuesByName) {
 				ISources.ACTIVE_WORKBENCH_WINDOW_SHELL_NAME, ISources.ACTIVE_EDITOR_ID_NAME,
 				ISources.ACTIVE_EDITOR_INPUT_NAME, ISources.SHOW_IN_INPUT, ISources.SHOW_IN_SELECTION,
 				ISources.ACTIVE_PART_NAME, ISources.ACTIVE_PART_ID_NAME, ISources.ACTIVE_SITE_NAME,
-				ISources.ACTIVE_CONTEXT_NAME, ISources.ACTIVE_CURRENT_SELECTION_NAME }));
+				ISources.ACTIVE_CONTEXT_NAME }));
```

Comment:
That part seems really strange. To be honest, I'd rather avoid just making `selection` an exception here without clear understanding of why this is necessary for it and not for others.

Reply:
Hopefully, Paul Webster can comment on that. First, I removed the whole blacklist but got many regressions, then limited to removing selection from it. I think since the bug exists since E4, a lot of people might have created workarounds and thus getting regressions when getting rid of the blacklist.

Reply:
I could imagine, that initially, those variables from IEvaluationContext collided with the variables from IEvaluationContext which are similar concepts but have slightly different semantics. So, in order to make both work together, the blacklist was introduced (just guessing).

Reply:
![image](https://user-images.githubusercontent.com/177225/205299875-da6ccd78-cf48-4076-888a-d23c41f2fd16.png)

at line 168, the denoted variable was filtered (2005)... it seems that filtering concept was then extended for many more variables in 2011 when compat layer was developed
that line became stale when the filter list was introduced

Reply:
@mickaelistria I do not think we will get any more confident about the intentions of that filter mechanism without a clearer hint... so, how to proceed?

Reply:
Looking at git blame, we may get some clues on bugs.eclipse.org/338056

Reply:
so, the change in that bug/commit suggests that the filtering mechanism was introduced to prevent source providers to overwrite "system variables" such as "selection"

Reply:
`org.eclipse.ui.ISourceProviderListener does` not mention that system variables are ignored...

Reply:
I believe the "system" source provider is a source provider like any other; so filtering its variables out here doesn't make sense: the EvaluationContext will just miss updates, and that -I believe- is not what is desired here.
I suspect you can try removing the variableFiler entirely and see what happens. It would be more correct to me.
About the issue of overriding variables or not, This is more something that the SourceProviderService should manage when loading the providers to enforce highest priority to the system ones; but as I see as a separate topic that we don't need to fix here, and that was neither necessary to fix as part of bugs.eclipse.org/338056

Reply:
I analyzed one failing test which already revealed a conceptual gap in the interaction between E3 and E4 compat layer:

1. the `org.eclipse.e4.ui.services.ContextServiceAddon` listens in the WorkbenchContext (IEclipseContext) for activeContexts changes. This is done at E4 level
2. the E3 compat layer listens for source changes in the IEvaluationService's context

Filtering those system variables makes sure that the IEvaluationService's context is not overwritten as it is kept up to date by E3 legacy mechanisms. The ContextServiceAddon (keeping the global ContextManager's state up to date) must be done in E3 compat layer not in E4 (as at E4 level there is no IEvaluationService) or at least make sure that the global ContextManager is fed by the state of the IEvaluationService.